test/sidechain/AuraBALOFT.spec.ts

```diff
@@ -0,0 +1,389 @@
+import { parseEther } from "@ethersproject/units";
+import { expect } from "chai";
+import { BigNumber, ContractTransaction, Signer } from "ethers";
+import hre, { ethers } from "hardhat";
+import { CanonicalPhaseDeployed, SidechainDeployed } from "scripts/deploySidechain";
+import {
+    getAura,
+    getAuraBal,
+    impersonateAccount,
+    increaseTime,
+    ONE_WEEK,
+    simpleToExactAmount,
+    ZERO,
+    ZERO_ADDRESS,
+} from "../../test-utils";
+import { Account } from "../../types";
+import { AuraBalOFT, AuraBalProxyOFT, AuraBalVault, ERC20, MockERC20__factory, ProxyOFT } from "../../types/generated";
+import shouldBehaveLikeERC20, { IERC20BehaviourContext } from "../shared/ERC20.behaviour";
+import { OwnableBehaviourContext, shouldBehaveLikeOwnable } from "../shared/Ownable.behaviour";
+import { SideChainTestSetup, sidechainTestSetup } from "./sidechainTestSetup";
+const NATIVE_FEE = simpleToExactAmount("0.2");
+const L1_CHAIN_ID = 111;
+const L2_CHAIN_ID = 222;
+const debug = false;
+async function bridgeTokenFromL1ToL2(
+    sender: Account,
+    token: ERC20,
+    proxyOFT: ProxyOFT,
+    dstChainId: number,
+    amount: BigNumber,
+): Promise<ContractTransaction> {
+    const from = sender.address;
+    const to = sender.address;
+    await token.connect(sender.signer).approve(proxyOFT.address, amount);
+    const tx = await proxyOFT
+        .connect(sender.signer)
+        .sendFrom(from, dstChainId, to, amount, ZERO_ADDRESS, ZERO_ADDRESS, [], {
+            value: NATIVE_FEE,
+        });
+    return tx;
+}
+describe("AuraBalOFT", () => {
```

Comment:
Can probably add some tests for `executeVault` here too.

Reply:
Done